client/container_top.go

```diff
@@ -4,17 +4,21 @@ import (
 	"context"
 	"encoding/json"
 	"net/url"
-	"strings"
 
 	"github.com/docker/docker/api/types/container"
 )
 
 // ContainerTop shows process information from within a container.
+// for `docker top redisstor -C "sto redis-server"`, we can't use string.Join(arguments, " ")
 func (cli *Client) ContainerTop(ctx context.Context, containerID string, arguments []string) (container.ContainerTopOKBody, error) {
 	var response container.ContainerTopOKBody
 	query := url.Values{}
 	if len(arguments) > 0 {
-		query.Set("ps_args", strings.Join(arguments, " "))
+		argsjson, err := json.Marshal(arguments)
+		if err != nil {
+			return response, err
+		}
+		query.Set("ps_args", string(argsjson))
```

Comment:
This is an API change, so (if we agree on this approach), must be documented in both the API reference, and API changelog. It should likely also be versioned (i.e., when using an older API version on a new daemon, it should behave the same as before, and don't accept the JSON format)

Reply:
Yes, I add a TODO in here after we agree on which version support this feature.
Thank you for your review.